pages/order.tsx

```diff
@@ -157,120 +165,101 @@ const Landing: NextPage = () => {
-      if (isFromBot) {
-        textAlign = "left";
-        textColor = "black";
+        dropdownWidth: '90%',
       }
-      switch (propName) {
-        case "listProps":
-          return ({
-              'left': "1px",
-              'marginLeft': textMarginSize,
-              'marginRight': textMarginSize,
-          });
-        case "textProps":
+
+      const getDropdownButton = (): React.ReactElement => {
+        if (specialRange) {
           return (
-            {
-              'textAlign': textAlign,
-              'color': textColor,
-              'whiteSpace': "pre-line",
-              // width: '80%',
-              'marginLeft': "2px",
-              'marginRight': "2px",
-            }
+            <StyledDiv>
+              <StyledSmallText isFromBot={isFromBot}>{`${text.substring(0, specialRange.begin - 1)} `}</StyledSmallText>
+              <SpecialText children={text.substring(specialRange.begin, specialRange.end + 1)} />
+              <StyledSmallText isFromBot={isFromBot}>{`${text.substring(specialRange.end + 1)} `}</StyledSmallText>
+            </StyledDiv>
           );
-        default:
-          return ({});
+        }
+        else {
+          return <SpecialText children={text} />;
+        }
       }
-    }
 
-  const createQRBubble = (QRFrame: React.ReactElement): React.ReactElement => {
-    return (
-      <TextBubbleContainer isFromBot={true}>
-        <TextBubble isFromBot={true}>{QRFrame}</TextBubble>
-      </TextBubbleContainer>
-    );
-  };
+      const getDropdownItems = (): React.ReactElement[] => {
+        return (
+          list.map((_, i) => (
+            <DropdownItem>
+                {list[i]}
+            </DropdownItem>
+          ))
+        );
+      }
+
+      return (
+        <StyledDropdown 
+          textMarginSize={textMarginSize} 
+          dropdownButton={getDropdownButton()}
+          {...listProps}
+        >
+          {getDropdownItems()}
+        </StyledDropdown>
+      );
 
-  const createTextBubble = (
-    highlightedString: HighlightedString
-  ): React.ReactElement => {
-    var icon = User;
-    const { isRight } = highlightedString;
-  
-    if (isRight) {
-      icon = Robot;
     }
 
     return (
-      <TextBubbleContainer isFromBot={!!isRight}>
-        <StyledImg isFromBot={!!isRight} as={icon} imgSize={iconSize} />
-        <TextBubble isFromBot={!!isRight}>
+      <TextBubbleContainer isFromBot={isFromBot}>
+        <StyledImg isFromBot={isFromBot} as={icon} imgSize={iconSize} />
+        <TextBubble isFromBot={isFromBot}>
           <StyledP textMarginSize={textMarginSize}>
-            <HighlightedText labels={[highlightedString]} />
+            {getTextBubble()}
           </StyledP>
         </TextBubble>
       </TextBubbleContainer>
     );
 
   };
 
-  const displayHighlightedText = (): React.ReactElement => {
-    return <>{highlightedStrings}</>;
+  const getTextBubbles = (): React.ReactElement => {
+    return <>{textBubbles}</>;
```

Comment:
Can't you just return without fragments?

Reply:
textBubbles is an array of React.ReactElements so it needs to be wrapped in a parent element. I will replace empty fragments with a div element.